File: Binary_Search/Binary_Search.java
// Binary Search in Java

class Binary_Search {
    int binarySearch(int array[], int x, int low, int high) {
  
      // Repeat until the pointers low and high meet each other
      while (low <= high) {
        int mid = low + (high - low) / 2;
  
        if (array[mid] == x)
          return mid;
  
        if (array[mid] < x)
          low = mid + 1;
  
        else
          high = mid - 1;
      }
  
      return -1;
    }
  // main
    public static void main(String args[]) {
      Binary_Search ob = new Binary_Search();
      int array[] = { 3, 4, 5, 6, 7, 8, 9 };
      int n = array.length;
      int x = 8;
      int result = ob.binarySearch(array, x, 0, n - 1);
      if (result == -1)
        System.out.println("Not found");
      else
        System.out.println("Element found at index " + result);
    }
  }

  // This code is contributed by Ali Barzegari Dahaj

File: Binary_Search/Binary_Search.cs
// Binary Search in C#
using System;

public class Binary_Search
{
    // An iterative binary search function.
    public static int BinarySearch(int[] arr, int low, int high, int x)
    {
        while (low <= high)
        {
            int mid = low + (high - low) / 2;

            // Check if x is present at mid
            if (arr[mid] == x)
                return mid;

            // If x greater, ignore left half
            if (arr[mid] < x)
                low = mid + 1;

            // If x is smaller, ignore right half
            else
                high = mid - 1;
        }

        // If we reach here, then element was not present
        return -1;
    }

    public static void Main(void)
    {
        int[] arr = { 2, 3, 4, 10, 40 };
        int x = 10;
        int n = arr.Length;
        int result = BinarySearch(arr, 0, n - 1, x);
        Console.WriteLine(result == -1
            ? "Element is not present in array"
            : $"Element is present at index {result}");
    }
}

// This code is contributed by Ali Barzegari Dahaj
